src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,100 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = (T[]) new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
```

Comment:
* primitive fields have default value
* `this` is redundant since there are no other variables with similar name 
```suggestion
    public ArrayList() {
        elementData = (T[]) new Object[DEFAULT_CAPACITY];
    }
```